src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -909,6 +923,76 @@ private JButton getCleanTableButton() {
 		return resetButton;
 	}
 
+	private JButton getValidateButton() {
+		validateButton = new JButton(MessageBundle.getMessage("angal.inventory.validate.btn"));
+		validateButton.setMnemonic(MessageBundle.getMnemonic("angal.inventory.validate.btn.key"));
+		if (inventory == null) {
+			validateButton.setEnabled(false);
+		}
+		validateButton.addActionListener(actionEvent -> {
+			if (inventory == null) {
+				MessageDialog.error(null, "angal.inventory.inventorymustsavebeforevalidate.msg");
+				return;
+			}
+			List<MedicalInventoryRow> invRowWithoutLot = inventoryRowSearchList.stream().filter(invRow -> invRow.getLot() == null).collect(Collectors.toList());
+			if (invRowWithoutLot.size() > 0) {
+				MessageDialog.error(null, "angal.inventory.allinventoryrowshouldhavelotbeforevalidate.msg");
+				return;
+			}
+			int reset = MessageDialog.yesNo(null, "angal.inventoryrow.doyoureallywanttovalidatethisinventory.msg");
+			if (reset == JOptionPane.YES_OPTION) {
+				String dischargeCode = inventory.getDischargeType();
+				String chargeCode = inventory.getChargeType();
+				Integer supplierId = inventory.getSupplier();
+				String wardCode = inventory.getDestination();
+				if (dischargeCode == null || dischargeCode.equals("")) {
+					MessageDialog.error(null, "angal.inventory.choosedischargetypebeforevalidate.msg");
+					return;
+				}
+				if (chargeCode == null || chargeCode.equals("")) {
+					MessageDialog.error(null, "angal.inventory.choosechargetypebeforevalidate.msg");
+					return;
+				}
+				if (supplierId == null || supplierId == 0) {
+					MessageDialog.error(null, "angal.inventory.choosesupplierbeforevalidate.msg");
+					return;
+				}
+				if (wardCode == null || wardCode.equals("")) {
+					MessageDialog.error(null, "angal.inventory.choosesupplierbeforevalidate.msg");
+					return;
+				}
+				// validate inventory
+				int inventoryRowsSize = inventoryRowSearchList.size();
+				try {
+					medicalInventoryManager.validateInventory(inventory, inventoryRowSearchList);					
+				} catch (OHServiceException e) {
+					OHServiceExceptionUtil.showMessages(e);
+				}
```

Comment:
If validation fails has messages means that has failed, so it should not close the inventory and set its status to VALIDATED, but rather go back to editing, so the user can check the updated quantities and new lots added to the list, before to proceed with a new validation

Reply:
I've done this just to inform the user that the theoretical quantity has changed because the validatInventory method in the manager returns nothing (void) and if I want to return a message to the user to tell him what has changed, how do I do this with the validateInvenoty method in the manager?

Reply:
after the catch, don't go ahead, and just `return;`